modules/sw360/sw360-workflow/src/main/java/org/eclipse/sw360/antenna/sw360/SW360MetaDataUpdater.java

```diff
@@ -108,15 +112,74 @@ public boolean isUploadSources() {
     public SW360Release uploadAttachments(SW360Release sw360Release, Map<Path, SW360AttachmentType> attachments) {
         AttachmentUploadRequest.Builder<SW360Release> builder = AttachmentUploadRequest.builder(sw360Release);
         for (Map.Entry<Path, SW360AttachmentType> e : attachments.entrySet()) {
-            builder = builder.addAttachment(e.getKey(), e.getValue());
+            if (needUpload(sw360Release, e.getKey(), e.getValue())) {
+                builder = builder.addAttachment(e.getKey(), e.getValue());
+            }
         }
 
-        AttachmentUploadResult<SW360Release> result = releaseClientAdapter.uploadAttachments(builder.build());
-        LOGGER.debug("Result of attachment upload operation: {}", result);
-        if (!result.isSuccess()) {
-            LOGGER.error("Failed to upload attachments: {}", result.failedUploads());
+        AttachmentUploadRequest<SW360Release> uploadRequest = builder.build();
+        if (!uploadRequest.getItems().isEmpty()) {
+            AttachmentUploadResult<SW360Release> result = releaseClientAdapter.uploadAttachments(uploadRequest);
+            LOGGER.debug("Result of attachment upload operation: {}", result);
+            if (!result.isSuccess()) {
+                LOGGER.error("Failed to upload attachments: {}", result.failedUploads());
+            }
+            return result.getTarget();
+        }
+        return sw360Release;
+    }
+
+    /**
+     * Calculates the SHA-1 hash for the given local attachment file. This is
+     * used to determine whether a modified attachment file needs to be
+     * uploaded.
+     *
+     * @param path the path to the local file
+     * @return the SHA-1 hash of this file
+     * @throws SW360ClientException if an error occurs
+     */
+    String calculateAttachmentHash(Path path) {
+        return SW360AttachmentUtils.calculateSha1Hash(path);
+    }
+
+    /**
+     * Checks whether an attachment upload is necessary for the given
+     * attachment file. This is the case if no attachment with this file name
+     * and type and identical content exists.
+     *
+     * @param release        the release affected
+     * @param path           the path to the local attachment file
+     * @param attachmentType the attachment type
+     * @return a flag whether this attachment file must be uploaded
+     */
+    private boolean needUpload(SW360Release release, Path path, SW360AttachmentType attachmentType) {
+        String fileName = String.valueOf(path.getFileName());
+        Optional<SW360SparseAttachment> optAttachment = release.getEmbedded().getAttachments().stream()
+                .filter(attachment -> attachmentMatches(path, attachmentType, fileName, attachment))
+                .findFirst();
```

Comment:
A duplicate attachment problem can also come from an attachment not related to this release. 

An additional possibility to check for the need to upload after this check could be to make a call <sw360-url>/attachments/?sha1=<sha1> and check for the response object. but that might be overdoing it.

Reply:
In my tests it was possible to upload the same attachment file with the same name to multiple releases. So I do not think that other releases affect a specific upload. Or did I get you wrong?

Reply:
I just checked the SW360 code again. You are correct. I did get it wrong.